Source/Core/VideoBackends/OGL/TextureCache.cpp

```diff
@@ -139,10 +141,62 @@ TextureCache::TCacheEntryBase* TextureCache::CreateTexture(const TCacheEntryConf
 
 void TextureCache::TCacheEntry::DoPartialTextureUpdate(TCacheEntryBase* entry_, u32 x, u32 y)
 {
-
 	TCacheEntry* entry = (TCacheEntry*)entry_;
-
-	glCopyImageSubData(entry->texture, GL_TEXTURE_2D_ARRAY, 0, 0, 0, 0, texture, GL_TEXTURE_2D_ARRAY, 0, x, y, 0, entry->native_width, entry->native_height, 1);
+	u32 w = native_width;
+	u32 h = native_height;
+	if (g_ActiveConfig.bCopyEFBScaled)
+	{
+		w = Renderer::EFBToScaledX(w);
+		h = Renderer::EFBToScaledY(h);
+	}
+	u32 max = g_renderer->GetMaxTextureSize();
+	if (max < w || max < h)
+	{
+		return;
+	}
+	if (!config.rendertarget || config.width != w || config.height != h)
+	{
+		GLuint temp_text = 0;
+		GLuint temp_fb = 0;
+		g_renderer->ResetAPIState();
+		glActiveTexture(GL_TEXTURE10);
+		glGenTextures(1, &temp_text);
+		glBindTexture(GL_TEXTURE_2D_ARRAY, temp_text);
+		glTexParameteri(GL_TEXTURE_2D_ARRAY, GL_TEXTURE_MAX_LEVEL, 0);
+		glTexImage3D(GL_TEXTURE_2D_ARRAY, 0, GL_RGBA, w, h, 1, 0, GL_RGBA, GL_UNSIGNED_BYTE, nullptr);
+		glGenFramebuffers(1, &temp_fb);
+		FramebufferManager::SetFramebuffer(temp_fb);
+		FramebufferManager::FramebufferTexture(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D_ARRAY, temp_text, 0);
+		glActiveTexture(GL_TEXTURE9);
+		glBindTexture(GL_TEXTURE_2D_ARRAY, texture);
+		g_sampler_cache->BindLinearSampler(9);
+		glViewport(0, 0, w, h);
+		s_ColorCopyProgram.Bind();
+		glUniform4f(s_ColorCopyPositionUniform, 0.0f, 0.0f,
+			(float)config.width, (float)config.height);
+		glDrawArrays(GL_TRIANGLE_STRIP, 0, 4);
```

Comment:
Do you think it's worth to use glBlitFramebuffer to save some gl calls? It might be faster on the GPU as there might be faster "kernels" than our shader. But then it also might hit a slow path in the driver with higher overhead :/